### ISSUE TITLE
[Resolved Bug] Unable to place Nanny, Farmer, Nanny Hitching post etc

### ISSUE THREAD

**Issue by Zedifuu** [VoidArkMods/QoLPlus#170]
## Quick Information
- **Server Operating System:** Windows - Build 33.79 (Installed mods: Level distribution API, QoL+, Winter Wonderland --passive)
- **Game Operating System:** Windows - Build 33.81 (CL 480619)
- **Mod Version:** 35 (4/2/24)

## What Happened?
We recently fresh installed this mod on our **The_Island ASA Dedicated Server**, As we missed the quality of life features brought by super structures. In-game when trying to place any of the QoL+ structures such as the Nanny, Farmer, Gardner, Hitching Post.. The placement ghost appears, the ghost is rotatable but when trying to place the structure the ghost fades away like the placement was cancelled. And the structure It remains in the inventory.

When reaching out to a server that we are pals with with they have the same server game and mod versions but had previously installed at previous versions of the mod and their structures are currently working, So it may be a fresh install issue.

There are no recorded exceptions raised in the server terminal. Other mod components such as the Omni-tool and transfer functionality is working as intended so only appears to be structure issues. 

**Please let us know if there is any additional information required for triage.** 

### Mod configuration block in Server `GameUserSettings.ini`
```
[QoLPlus]
DisableHitchingPostMatingBonus=True
HitchingPostDinoLimit=3
HitchingPostTribeLimit=4
QoLPlusEngramWhitelist=EngramEntry_SplusMultiTool_C,EngramEntry_OmniTool_C,EngramEntry_Gardener_C,EngramEntry_Farmer_C,EngramEntry_Nanny_C,EngramEntry_HitchingPost_C
DisableMultiToolDinoChibiMode=True
OmniToolBlacklist=InventoryAssistant
```

## Expected result
When placing the structure it should place normally. 

## Screenshots
https://github.com/VoidArkMods/QoLPlus/assets/73708675/0fe41f55-aea3-4e8d-be01-4651bc7f83a6

## Estimated Priority/Severity
- [x] High (anything that impacts the normal user flow or blocks app usage)
- Medium (anything that negatively affects the user experience)
- Low (anything else e.g., typos, missing icons/translations, layout/formatting issues, etc.)

## Similar issues: https://github.com/VoidArkMods/QoLPlus/issues/130
On the above linked issue user dash74 [has mentioned a workaround](https://github.com/VoidArkMods/QoLPlus/issues/130#issuecomment-1890832159) was to remove mods in 'passive mod' such as Winter Wonderland, Which is currently in effect on our instance in passive mode. But we are unsure whether it is safe to remove this mod yet without losing the Christmas items so we are hesitant to remove this currently.

**Comment by VoidArkMods** [comment]
Hello, and thanks for all the information, I wish every opened issue was that descriptive!

When the item appears, the preview structure is there but they cannot be placed, this usually means the mod did not install itself properly on the server-side but did install properly on the client-side. To be safe, I suggest that you re-install the mod in both places.
The only reliable way of reinstalling a mod that I know of is to browse to the mod's install directory and delete the folder. QoL+ is very small in space, it can reinstall very quickly.
On a dedicated server, browse to `\ShooterGame\Binaries\Win64\ShooterGame\Mods\83374\939228_5078487` and delete the folder `QoLPlus`. The next time the server will start, the server version of the mod will be downloaded fresh, hopefully fixing any corruption. As the already placed structures, learnt engrams and items are stored at the map or player lever, this will not destroy anything even if the mod was already installed for a while.
On a client, browse to `C:\Program Files (x86)\Steam\steamapps\common\ARK Survival Ascended\ShooterGame\Binaries\Win64\ShooterGame\Mods\83374\939228_5078484` and delete the `QoLPlus` folder. Once again, doing so will not destroy structures/items/progression in your saved single player game.

There is also a known bug that makes the mod behave unexpectedly when it is the first one in the load order. I'm working on fixing that, but as a workaround you can try placing any other mod in the first spot. QoL+ does not require to be the first mod in the list.

**Comment by Zedifuu** [comment]
No worries, I maintain a few projects so I feel you 😅 also Thank you for the rapid response and detailed steps to resolution <3 

So TL;DR issue resolved 🎉 
We tried both suggested fixes so unfortunately cant determine which one was the specific cause, The mod was indeed the first in the list to be loaded so we pushed it to the end of the list as well as reinstalling server side with the method described and that did the trick :) (Clients did not require reinstall)

Again thanks for the clean advice and your continued development of this mod :) 

Also side observation which is possibly related, when the mod was installed in an unhealthy state, every player's character had their hair reset to a black afro or bald for some (which was quite a surprise) after resolving the above issue, users hair was also reverted to what it previously was. How strange 😂

**Comment by Zedifuu** [comment]
Side note @VoidArkMods , If you are interested and have some spare time you could maybe look at [GitHub's Beta Issue Template](https://docs.github.com/en/communities/using-templates-to-encourage-useful-issues-and-pull-requests/configuring-issue-templates-for-your-repository#creating-issue-forms) feature. Could also setup some fancy ci to update the version selector array and commit it for you too ✨

I'm not sure how fleshed out it is but if it works, it may improve the issue quality you receive on this project, we found issue templates on Gitlab prove a substantial increase in issue quality 😁

**Comment by VoidArkMods** [comment]
I'll keep an eye on that. The thing is, as the issue tracker is the only available way to provide feedbacks (as comments are disabled on CurseForge), only a few users are used to how issue trackers work. If this can help increasing the amount of information in new issues, reduce the duplicates and overall increase the experience both on the users side and the dev side, that would be really interesting.

Regarding the hair issue, there were a couple of mistakes during the migration from S+ to QoL+ (my bad). Some of them have been fixed, others are just detected but not fixed yet, others are a bit low on the priority list to have some focus time for now, and some may have not even been detected. The hair issue has been encountered by a couple of users, it can be reproduced, but unfortunately I didn't have the time to focus on it yet. It is likely related to the main file of the mod which changed a lot between ASE and ASA, but it's risky to play with it as everything is referenced in this single file. When the important features and bugs are done and fixed, this will get fixed.

**Comment by Zedifuu** [comment]
Gotcha gotcha! Maybe pop into a private repo and have a play with it, see if it's something that'd fit your needs, maybe make 
 it a bit simpler than what I templated above, could also omit the version field if that's more of a Curseforge specific variable or make it like a text input field. Users 'should' always be on latest, but it may catch some who have not updated a latest release. Saves a discovery step but may be unnecessary. 

Oh man the mod is all in one big monolithic file? Kudo's to you! That sounds awful to deal with, I've had similar experiences in my work dealing with huge mission critical monolithic apps made by a person who left the company years ago and no ones maintained it since, leaving it to fester. 🥲 The hair issue has appeared to have resolved in any case so we're not too bothered, worst comes to the worst we can just use scissors

**Comment by Zedifuu** [comment]
If you want to get REAL fancy, you could look at at build automation, as there looks to be github actions out there for release automation to curseforge too :) Save you some manual steps. I'd offer my assistance as a DevOps Eng, but I would need the build process outlined as I'm quite unfamiliar with the UE build process 😅

Build Release > Add release to Github Releases incl changelog > [Push to Curseforge (example)](https://github.com/Funwayguy/BetterQuesting/pull/756/files#diff-de845b643c7c5d83c58e1d48427ec9a5429096687fca59cae7043045bbfef9a4R35-R46)

https://github.com/marketplace/actions/upload-to-curseforge
